tutorial/20_recipes/006_user_defined_pruner.py

```diff
@@ -4,24 +4,53 @@
+through the :meth:`study.get_trial` method and, and from a trial, its reported 
+intermediate values through the `trial.intermediate_values` attribute (a 
+dictionary which maps an integer `step` to a float value).
+
+You can refer to the source code of the built-in Optuna pruners as templates for 
+building your own. In this document, for illustration, we describe the 
+construction and usage of a simple (but aggressive) pruner which prunes trials 
+that are in last place compared to completed trials at the same step.
 
 .. note::
-  Please refer to the documentation of :class:`~optuna.pruners.BasePruner` or, for example, :class:`~optuna.pruners.ThresholdPruner` or :class:`optuna.pruners.PercentilePruner` for more robust examples of pruner implementation, including error checking and complex pruner-internal logic.
+  Please refer to the documentation of :class:`~optuna.pruners.BasePruner` or, 
+  for example, :class:`~optuna.pruners.ThresholdPruner` or 
+  :class:`optuna.pruners.PercentilePruner` for more robust examples of pruner 
+  implementation, including error checking and complex pruner-internal logic.
 
 An Example: Implementing LastPlacePruner
 --------------------------------------------------
 
-We aim to optimize the `loss` and `alpha` hyperparameters for a stochastic gradient descent classifier (SGDClassifier) run on the sklearn iris dataset. We implement a pruner which terminates a trial at a certain step if it is in last place compared to completed trials at the same step. We begin considering pruning after a "warmup" of 1 training step and 5 completed trials. For demonstration purposes, we print() a diagnostic message from prune() when it is about to return True (indicating pruning). 
-
-It may be important to note that the SGDClassifier score, as it is evaluated on a holdout set, decreases with enough training steps due to overfitting. This means that a trial could be pruned even if it had a favorable (high) value on a previous training set. After pruning, Optuna will take the intermediate value last reported as the value of the trial.
+We aim to optimize the `loss` and `alpha` hyperparameters for a stochastic 
+gradient descent classifier (SGDClassifier) run on the sklearn iris dataset. We 
+implement a pruner which terminates a trial at a certain step if it is in last 
+place compared to completed trials at the same step. We begin considering 
+pruning after a "warmup" of 1 training step and 5 completed trials. For 
+demonstration purposes, we print() a diagnostic message from prune() when it is 
+about to return True (indicating pruning). 
+
+It may be important to note that the SGDClassifier score, as it is evaluated on 
+a holdout set, decreases with enough training steps due to overfitting. This 
+means that a trial could be pruned even if it had a favorable (high) value on a 
+previous training set. After pruning, Optuna will take the intermediate value 
+last reported as the value of the trial.
```

Comment:
This line references a programming decision made at:
https://github.com/optuna/optuna/blob/35cc4c14471b501f1c76975bd587448fbe70c8ce/optuna/_optimize.py#L202